cloudsync_onedrive.py

```diff
@@ -800,9 +802,11 @@ def rename(self, oid, path):  # pylint: disable=too-many-locals, too-many-branch
                 self.rename(oid, path)
 
         new_path = self._get_path(oid)
-        if new_path != path:
-            log.error("path mismatch after rename -- wanted %s, got %s", path, new_path)
-            assert new_path == path  # must raise, cuz it's in the if block
+        new_dir, new_file = self.split(new_path)
+        old_dir, old_file = self.split(old_path)
+        if self.paths_match(old_dir, new_dir) and old_file == new_file:
+            log.error("old and new paths match after rename: %s", old_path)
+            raise CloudTemporaryError("old and new paths match after rename")
```

Comment:
I'm don't think the issue is that old and new paths match after rename, this matching is just how you know the issue exists. The issue is that the file was never actually renamed.

Reply:
true, the messaging could be clearer -- will adjust